tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Grpc/GrpcDotNet/GrpcAspNetCoreServer/GrpcDotNetServerCommon.cs

```diff
@@ -21,15 +21,31 @@ internal static class GrpcDotNetServerCommon
     {
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(GrpcDotNetServerCommon));
 
+        public static bool IsASupportedVersion<TTarget>()
+        {
+            return SupportedVersionByTypeCache<TTarget>.IsSupported;
+        }
+
         public static Scope? CreateServerSpan<T>(Tracer tracer, T instance, HttpRequest requestMessage)
         {
             if (!tracer.Settings.IsIntegrationEnabled(IntegrationId.Grpc))
             {
                 return null;
             }
 
+            // Check if the current handler has the MethodInvoker property
+            MethodStruct method;
+            if (instance.TryDuckCast<ServerCallHandlerBaseStruct>(out var handlerWithMethodInvoker))
```

Comment:
We don't need this any more with the version check above do we?

Reply:
True, I leave it as a good practice, Try > Cast

Reply:
For some reason I thought Try was less efficient, is that not the case? 🤔

Reply:
pretty much the same, the Try just add a check to the `Success` property of the CreateProxyResult before trying to create the instance.